README.md

<!-- README.md is generated from README.Rmd. Please edit that file -->
<!-- badges: start -->
<!-- badges: end -->

# Reproducible Climate Futures (rcf)

## Overview

This package aims to make acquiring and working with [MACA
v2](http://www.climatologylab.org/maca.html) climate data faster and
easier and to provide a number of summary statistics that can be used to
visualize different climate futures. Ultimately, having access to this
data supports planning efforts that aim to incorporate climate change.

## Installation

Until approval on CRAN, you can download the development version of
`rcf`

You can install the released version of rcf from
[CRAN](https://CRAN.R-project.org) with:

``` r
install.packages("rcf")
```

And the development version from [GitHub](https://github.com/) with:

``` r
# install.packages("devtools")
devtools::install_github("nationalparkservice/rcf")
```

``` r
library(tidyverse)
#> -- Attaching packages --------------------------------------- tidyverse 1.3.1 --
#> v ggplot2 3.3.5     v purrr   0.3.4
#> v tibble  3.1.3     v dplyr   1.0.7
#> v tidyr   1.1.3     v stringr 1.4.0
#> v readr   2.0.1     v forcats 0.5.1
#> -- Conflicts ------------------------------------------ tidyverse_conflicts() --
#> x dplyr::filter() masks stats::filter()
#> x dplyr::lag()    masks stats::lag()
library(rcf)
```

## Usage

Download data using the `rcf_data()` function from the `rcfdata` package
to start visualizing climate futures

``` r
# devtools::install_github("nationalparkservice/rcfdata")
# library(rcfdata)
# raw_data <- rcf_data(SiteID = "BAND",
#                      latitude = 35.75758546,
#                      longitude = -106.3054344,
#                      units = "imperial")
```

``` r
raw_data <- read_csv("https://irmadev.nps.gov/DataStore/DownloadFile/660685")
#> Rows: 2191480 Columns: 10
#> -- Column specification --------------------------------------------------------
#> Delimiter: ","
#> chr  (2): gcm, units
#> dbl  (7): yr, precip, tmin, tmax, tavg, rhmin, rhmax
#> dttm (1): date
#> 
#> i Use `spec()` to retrieve the full column specification for this data.
#> i Specify the column types or set `show_col_types = FALSE` to quiet this message.
```

Calculate threshold values using `calc_thresholds()` and summarize them
by month, season or year as well as by quadrant or the most extreme
model in each quadrant using `cf_quadrant()`.

``` r
thresholds <- calc_thresholds("BAND", data = raw_data, units = "imperial")
#> Adding missing grouping variables: `gcm`
#> Warning in calc_thresholds("BAND", data = raw_data, units = "imperial"): Files
#> have been saved to temporary directory and will be deleted when this R session
#> is closed. To save locally, input a local directory in which to save files into
#> the `directory` argument.
#> Warning in calc_thresholds("BAND", data = raw_data, units = "imperial"):
#> thresholds.csv generated successfully. DO NOT edit this csv in excel. File is
#> too large and data will be lost, causing errors in future calculations.

quadrant_year <- cf_quadrant("BAND", data = thresholds, future_year = 2040, summarize_by = "year", method = "quadrant")
#> Warning in cf_quadrant("BAND", data = thresholds, future_year = 2040,
#> summarize_by = "year", : Files have been saved to temporary directory and
#> will be deleted when this R session is closed. To save locally, input a local
#> directory in which to save files into the `directory` argument.
```

From here we can use `ggplot2` to visualize any variables and how they
compare between the 4 climate futures as well as between past and
future.

``` r
quadrant_year_future <- quadrant_year %>%
filter(time %in% c("Future"))

ggplot(data = quadrant_year_future, aes(x = cf, y = freeze_thaw)) +
geom_boxplot(alpha = 0.4,
             aes(color = cf, fill = cf)) +
geom_jitter(alpha = 0.7,
            aes(color = cf, fill = cf)) +
scale_fill_viridis_d() +
scale_color_viridis_d() +
  labs(y = "Number of days per year",
       title = "Days that have a freeze thaw cycle by climate future") +
  theme(axis.title.x = element_blank()) +
  theme_minimal()
```

<img src="man/figures/README-unnamed-chunk-6-1.png" width="110%" />

## Explore further

For a more in-depth explanation of the `rcf` package and different ways
to download the data, you can follow along with [An Introduction to the
Reproducible Climate Futures
package](https://github.com/Janelle88/rcf_addendum/blob/master/RCF%20instructions%20-%20simple.pdf).

## Data

<img src="man/figures/kable_table.png" width="90%" style="display: block; margin: auto;" />


NEWS.md
# rcf version 0.0.1

Initial submission to CRAN


test_script.R
## script to test package

my_directory <- "C:/Users/jnchr/Documents/test/SCBL"

rcf_data("SCBL",latitude = 41.83476, longitude = -103.707, directory = my_directory)


raw_data <- readr::read_csv('C:/Users/jnchr/Documents/test/BAND.csv')

# band_small <- raw_data %>%
#   dplyr::filter(gcm %in% c("bcc-csm1-1.rcp45", "bcc-csm1-1.rcp85", "BNU-ESM.rcp45", "BNU-ESM.rcp85",
#                            "CanESM2.rcp45", "CanESM2.rcp85", "CCSM4.rcp45", "CCSM4.rcp85",
#                            "GFDL-ESM2G.rcp45", "GFDL-ESM2G.rcp85"))
#
# band_small <- dplyr::sample_n(band_small, 2500)
#
# readr::write_csv(band_small, "C:/Users/jnchr/Documents/test/BAND_smaller.csv")

# small_data <- dplyr::sample_n(raw_data, 10000)

# raw_data <- small_data

# readr::write_csv(small_data, "C:/Users/jnchr/Documents/R/rcf/data/BAND_small.csv")

# ----------------
# CALC THRESHOLDS
# ----------------

calc_thresholds("BAND", data = raw_data, directory = my_directory, units = "imperial")

thresholds <- readr::read_csv('C:/Users/jnchr/Documents/test/BAND_test_thresholds.csv')

# ----------------
# CF QUADRANT
# ----------------

# data <- thresholds

cf_quadrant("BAND_test", data = thresholds, future_year = 2040, summarize_by = "year", method = "quadrant", directory = my_directory)

cf_quadrant("BAND_test", data = thresholds, future_year = 2040, summarize_by = "month", method = "quadrant", directory = my_directory)

cf_quadrant("BAND_test", data = thresholds, future_year = 2040, summarize_by = "season", method = "quadrant", directory = my_directory)

cf_quadrant("BAND_test", data = thresholds, future_year = 2040, summarize_by = "year", method = "corner", directory = my_directory)

cf_quadrant("BAND_test", data = thresholds, future_year = 2040, summarize_by = "month", method = "corner", directory = my_directory)

cf_quadrant("BAND_test", data = thresholds, future_year = 2040, summarize_by = "season", method = "corner", directory = my_directory)

# -------------
# SUMMARIZE FOR PCA
# -------------

summarize_for_pca("BAND_test", data = thresholds, future_year = 2040, directory = my_directory)

pca_summary <- readr::read_csv('C:/Users/jnchr/Documents/test/BAND_pca_summary.csv')

# -------------
# CF PCA
# -------------

data <- pca_summary

cf_pca("BAND_test", data = pca_summary, variables = "all_threshold", directory = my_directory)

pca_data <- readr::read_csv('C:/Users/jnchr/Documents/test/BAND_test_future_means_pca.csv')

# -------------
# PCA THRESHOLDS
# -------------

# data <- pca_data
# all_data <- thresholds

pca_thresholds("BAND_test", pca_data = pca_data, all_data = thresholds, summarize_by = "year", directory = my_directory)

pca_thresholds("BAND_test", pca_data = pca_data, all_data = thresholds,  summarize_by = "month", directory = my_directory)

pca_thresholds("BAND_test", pca_data = pca_data, all_data = thresholds,  summarize_by = "season", directory = my_directory)

# variables if needed
# year = 2040
# rh_exists <-  any(names(data) == "rhmin")
# variables = c("precip", "tmin", "tmax", "tavg", "rhmin", "rhmax", "heat_index_ec", "heat_index_dan", "temp_over_95_pctl", "temp_over_99_pctl", "temp_over_95_pctl_length", "temp_under_freeze", "temp_under_freeze_length", "temp_under_5_pctl", "no_precip" , "no_precip_length", "precip_95_pctl", "precip_99_pctl", "precip_moderate", "precip_heavy", "freeze_thaw", "gdd", "gdd_count", "not_gdd_count", "frost", "grow_len")


tests/testthat.R
library(testthat)
library(rcf)

test_check("rcf")


tests/testthat/test_cf_pca.R
data_for_pca <- readr::read_csv(system.file("extdata", "BAND_pca_summary.csv", package = "rcf"))

data <- readr::read_csv(system.file("extdata","BAND_small.csv", package = "rcf"))

my_directory <- here::here()

test_that("Function results in a dataframe", {
  skip_on_cran()
  pca_means <- cf_pca(SiteID = "BAND",
                               data = data_for_pca,
                               variables = "all_threshold",
                               num_cf = 4,
                               directory = my_directory)
  expect_s3_class(pca_means, "data.frame")
})

test_that("num_cf is 2 or 4", {
  expect_error(cf_pca(SiteID = "BAND",
                      data = data_for_pca,
                      variables = "all_threshold",
                      num_cf = 3,
                      directory = my_directory))
})

testthat::test_that("Variable names match column names", {
  testthat::expect_error(
    cf_pca(SiteID = "BAND",
           data = data,
           variables = c("cat", "dog"),
           directory = my_directory))
})


test_that("tempdir() gives warning", {
  expect_warning(cf_pca(SiteID = "BAND",
                        data = data_for_pca,
                        variables = "all_threshold"))
})


R/cf_pca.R
#' Calculate climate futures using PCA
#'
#' Designates climate futures of "Warm Wet", "Warm Dry", "Hot Wet", "Hot Dry" and "Central"
#' and calculates mean values for each GCM. Will additionally calculate which models
#' represent the largest spread in data of variables entered using PCA. Calculates
#' summary of threshold values based off of selected summarization parameter and PCA.
#'
#' *For advanced users only.
#'
#' @param SiteID chosen name to use in file names, attributes, and directories. Default
#' name is "unnamed_site" (character)
#' @param data Default dataset to use for the .csv files this function will create.
#' Follow vignette for example dataset creation. (data frame)
#' @param variables Variables you want the PCA to be based off of. Must match column names
#' from dataframe in `data` parameter exactly. If running directly from
#' `summarize_for_pca` and would like to use all threshold values to select models,
#' write "all_threshold" (character)
#' @param num_cf Number of climate futures to select. Option of 2 or 4. Two models will be the
#' maximum and minimum values of principal component 1 (PC1), and 4 will be the maximum
#' and minimum values of principal component 2 (PC2)
#' @param directory where to save files to. Per CRAN guidelines, this
#' defaults to a temporary directory and files created will be lost after
#' R session ends. Specify a path to retain files.
#'
#' @return
#' One (1) csv file that has selected either 2 or 4 models, depending on `num_cf` chosen
#' based upon PCA
#' One (1) png file of the PCA scatterplot, with models labeled
#'
#' @export
#'
#' @examples
#'
#' \dontrun{
#' # Generate sample data
#'
#' data <- data.frame(
#'  gcm = c("bcc-csm1-1.rcp45", "BNU-ESM.rcp45", "CanESM2.rcp85", "CCSM4.rcp45",
#'              "CSIRO-Mk3-6-0.rcp45"),
#'  precip_change = rnorm(5),
#'  tmin_change = rnorm(5),
#'  tmax_change = rnorm(5),
#'  rhmax_change = rnorm(5),
#'  rhmin_change = rnorm(5),
#'  tavg_change = rnorm(5),
#'  heat_index_change = rnorm(5),
#'  heat_index_ec_change = rnorm(5),
#'  heat_index_dan_change = rnorm(5),
#'  temp_over_95_pctl_change =  rnorm(5),
#'  temp_over_99_pctl_change =  rnorm(5),
#'  temp_over_95_pctl_length_change =  rnorm(5),
#'  temp_under_freeze_change =  rnorm(5),
#'  temp_under_freeze_length_change =  rnorm(5),
#'  temp_under_5_pctl_change =  rnorm(5),
#'  no_precip_change  =  rnorm(5),
#'  no_precip_length_change =  rnorm(5),
#'  precip_95_pctl_change =  rnorm(5),
#'  precip_99_pctl_change =  rnorm(5),
#'  precip_moderate_change =  rnorm(5),
#'  precip_heavy_change =  rnorm(5),
#'  freeze_thaw_change =  rnorm(5),
#'  gdd_change =  rnorm(5),
#'  gdd_count_change = rnorm(5),
#'  not_gdd_count_change = rnorm(5),
#'  frost_change = rnorm(5),
#'  grow_length_change = rnorm(5),
#'  units = rep("imperial", each = 5)
#' )
#'
#' cf_pca("SCBL", data = data, variables = c("tmin", "tmax", "rhmin"),
#' num_cf = 2)
#'}
#'
#' @importFrom magrittr %>%
#' @importFrom rlang .data
#' @import ggfortify
#' @importFrom utils globalVariables
#'
#' @name cf_pca

utils::globalVariables("where")

cf_pca <- function(SiteID = "unnamed_site",
                   data = NULL,
                   variables = "all_threshold",
                   num_cf = 4,
                   directory = tempdir()){

  first_variable = variables[1]

  is_all_threshold <- ifelse(first_variable == "all_threshold", TRUE, FALSE)

  all_threshold = c("precip_change", "tmax_change", "tmin_change",  "tavg_change", "rhmin_change", "rhmax_change", "heat_index_change", "heat_index_ec_change", "heat_index_dan_change", "temp_over_95_pctl_change", "temp_over_99_pctl_change", "temp_over_95_pctl_length_change", "temp_under_freeze_change", "temp_under_freeze_length_change", "temp_under_5_pctl_change", "no_precip_change", "no_precip_length_change", "precip_95_pctl_change", "precip_99_pctl_change", "precip_moderate_change", "precip_heavy_change", "freeze_thaw_change", "gdd_change", "frost_change", "grow_length_change")

    function_variables = if(is_all_threshold == TRUE){paste(all_threshold)} else{paste(variables)}

  #if() else() didn't work here - I've gone back and forth between the two functions, both
  # ultimately throw the error "the condition has length > 1 and only the first
  # element will be used" cannot say why
  #create variables to be used for the function

  # stop function with error messages

  if(num_cf %in% c(2,4) == FALSE){
    stop("num_cf can only be 2 or 4. Did you enter a different number?")
  }

variables_df <- data %>%
  dplyr::select(where(is.numeric))

#!sum(variables %in% variables_df) == length(variables)

  if(!sum(function_variables %in% colnames(variables_df)) == length(function_variables)){
    #number of matches for column names in variables argument does not match the number of
    #variables entered, then a variable has been entered incorrectly
    stop("Variable names entered are not the same as the column names of your dataframe. Please check for spelling or case sensitive errors. You also may have entered variables that contain non-numeric values.")
  }


  suppressMessages(if(!file.exists(".here")) here::set_here(directory))

  # ---------
  # subset data from entered dataframe
  # ---------

  future_all <- data

  named_df2 <- data.frame(tibble::column_to_rownames(data, var = "gcm")) %>%
    dplyr::select(base::paste(function_variables))
    # give rownames to variables for PCA
  named_df1 <- named_df2 %>%
    dplyr::select_if(!is.na(base::colSums(named_df2)) == "TRUE")
  # remove columns with NA values

  # needs to be split into two statements because it's looking for the
  # number of columns in the original dataset and throws an error

  named_df <- named_df1 %>%
    dplyr::select_if(!base::colMeans(named_df1, na.rm = TRUE) == base::unlist(named_df1[1,]))
    #remove columns where the mean is the same as the first value

  # ---------
  # Create PCA
  # ---------

  cf_pca <- named_df %>% #remove columns with NA values
    dplyr::select_if(~ !any(is.na(.data))) %>%
    # remove columns with NA values
    stats::prcomp(center = TRUE, scale. = TRUE)

  #print warning for removed columns

  kept_columns <- data %>%
    dplyr::select_if(~ !any(is.na(.data)))

  removed_columns <- data %>%
    dplyr::select(!colnames(kept_columns))

  columns_empty <- length(colnames(removed_columns) == 0)

  if(columns_empty > 0L)print(paste("Removed", colnames(removed_columns), "due to NA or all duplicate values in column.", colnames(removed_columns), "was not used in PCA calculation."))




  # ------------
  # PCA SCATTERPLOT
  # ------------

   pca_plot <- ggplot2::autoplot(cf_pca,
                    data = named_df,
                    loadings = TRUE,
                    loadings.label = TRUE,
                    label = TRUE) +
    ggplot2::theme_classic() + #has the L shape around the graph
    ggplot2::theme(panel.grid = ggplot2::element_blank(), #removes grid lines
                               plot.title = ggplot2::element_text(size = 25), #title of plot text size
                               legend.text = ggplot2::element_text(size = 15), #legend inside text size
                               legend.title = ggplot2::element_text(size = 18), #legend title text size
                               axis.title.y = ggplot2::element_text(size = 15), #changes y axis lable text size
                               axis.title.x = ggplot2::element_text(size = 15),#x axis text size
                               axis.text.x = ggplot2::element_text(size = 15), #changes x-axis text size
                               axis.text.y = ggplot2::element_text(size = 15), #changes y-axis text size
                               legend.position = "top")  #location of legend

  if(directory == tempdir()){warning("Files have been saved to temporary directory and will be deleted when this R session is closed. To save locally, input a local directory in which to save files into the `directory` argument.")}

  ggplot2::ggsave(here::here(directory,
                             paste0(SiteID, "_pca.png")),
                             width = 15, height = 9)



  # ---------------
  # MODEL SELECTION
  # ---------------

  pca_named<-as.data.frame(cf_pca$x)

  pc1_min <- rownames(pca_named)[which.min(pca_named$PC1)]
  pc1_max <- rownames(pca_named)[which.max(pca_named$PC1)]
  pc2_min <- rownames(pca_named)[which.min(pca_named$PC2)]
  pc2_max <- rownames(pca_named)[which.max(pca_named$PC2)]


  pc_models <- data.frame(pc1_max, pc1_min, pc2_max, pc2_min) %>%
    dplyr::rename(`PC1 Max` = .data$pc1_max,
                  `PC1 Min` = .data$pc1_min,
                  `PC2 Max` = .data$pc2_max,
                  `PC2 Min` = .data$pc2_min) %>%
    tidyr::pivot_longer(.data$`PC1 Max`:.data$`PC2 Min`,
                        names_to = "pca_type",
                        values_to = "gcm") %>%
    dplyr::filter(ifelse(num_cf == 2,
                         .data$pca_type %in% c("PC1 Max", "PC1 Min"),
                         .data$pca_type %in% c("PC1 Max", "PC1 Min", "PC2 Max", "PC2 Min")))

  pca_cf_gcm <- future_all %>%
    dplyr::full_join(pc_models, by = "gcm")

  readr::write_csv(pca_cf_gcm, here::here(directory,
                                          paste(SiteID,
                                                "future_means_pca.csv",
                                                sep = "_")))

  return(pca_cf_gcm)

}


R/rcf_data.R
#' Download MACA data from the cft package
#'
#' Downloads data from the cft package (https://www.earthdatascience.org/cft/)
#' to your parent directory.
#'
#' @param SiteID  chosen name to use in file names, attributes, and
#' directories. (character)
#' @param latitude latitude of point of interest (spatial)
#' @param longitude longitude of point of interest (spatial)
#' @param units the unit type that will be used,
#' defaults to "imperial" ("imperial" or "metric")
#' @param directory where to save files to. Per CRAN guidelines, this
#' defaults to a temporary directory and files created will be lost after
#' R session ends. Specify a path to retain files.
#'
#' @return one data frame - SiteID.csv - to be used
#' with other functions in this package(tibble)
#' @export
#'
#' @examples
#' \dontrun{
#' rcf_data(SiteID = "SCBL", latitude = 41.83476, longitude =  -103.707)
#'}
#'
#' @importFrom magrittr %>%
#' @importFrom rlang .data

rcf_data <- function(SiteID = "unnamed_site",
                     latitude,
                     longitude,
                     units = "imperial",
                     directory = tempdir()){

  suppressMessages(if(!file.exists(".here")) here::set_here(directory))

  ## Download data
  #Variable and scenario names corresponding to MACA data directory structure
  vars = c("pr", "tasmax", "tasmin","rhsmax","rhsmin")
  scens = c("rcp45", "rcp85")

  # gcms to be extracted
  gcms = c('bcc-csm1-1','bcc-csm1-1-m','BNU-ESM','CanESM2','CCSM4','CNRM-CM5','CSIRO-Mk3-6-0',
           'GFDL-ESM2G','GFDL-ESM2M','HadGEM2-CC365','HadGEM2-ES365',
           'inmcm4','IPSL-CM5A-MR','IPSL-CM5A-LR','IPSL-CM5B-LR',
           'MIROC5','MIROC-ESM','MIROC-ESM-CHEM','MRI-Cgcm3','NorESM1-M')

  #Date ranges to be extracted
  Future_StartYear = 2006   #2006-2099
  Future_EndYear = 2099   #2006-2099
  Hist_StartYear = 1950     #1950-2005
  Hist_EndYear = 2005      #1950-2005

  Remove_files = "Y" # "N"       #Removes all climate data files saved in directory

  ############################## END INITIALS ##################################################

  # Now can only use spatial object (not park name)
  Site_coordinates <- data.frame(PARK=SiteID,lat=latitude,lon=longitude)
  sp::coordinates(Site_coordinates) <- ~lon+lat
  sp::proj4string(Site_coordinates) <- "+proj=longlat +datum=NAD83 +no_defs " #same proj4string used in NPS_boundary_centroids.shp


  # # Load Data - from centroids
  # nps_boundary_centroids <- st_read('C:/Users/achildress/OneDrive - DOI/Documents/GIS/nps_boundary_centroids/nps_boundary_centroids.shp')
  # centroid <- filter(nps_boundary_centroids, UNIT_CODE == "BIBE")
  # Sp_centroid <- as_Spatial(centroid) # Does not accept sf objects

  # download data
  file_refs <- cft::cftdata(aoi = Site_coordinates,
                            area_name = SiteID,
                            years = c(Hist_StartYear, Future_EndYear),
                            models = gcms,
                            local_dir = directory,
                            parameters = vars,
                            scenarios = scens,
                            ncores = parallel::detectCores() / 2)

  df1 <- cft::cft_df(file_refs, ncores = parallel::detectCores() / 2)

  ######################## MANIPULATE INTO DF FOR PLOT_TABLE SCRIPT #####################
  if(units == "imperial"){

    df <- df1 %>%
      dplyr::mutate(precip = .data$pr/25.4,
                    # data source needs to be specified using rlang::.data
                    # error `no visible binding for global variable x` will be thrown
                    tmax = .data$tasmax * (9/5) - 459.67,
                    # decided to not label as tmax_f because it could cause conflicts
                    # if want to name as tmax_f will need to incorporate if statements
                    # to not cause any issues down the line if users choose "metric"
                    tmin = .data$tasmin * (9/5) - 459.67,
                    tavg = (.data$tmax + .data$tmin) / 2,
                    gcm = paste(.data$model, .data$rcp, sep = "."),
                    date = as.POSIXlt(.data$date,format="%Y-%m-%d"),
                    yr = lubridate::year(date),
                    units = "imperial") %>%
      dplyr::rename(rhmax = .data$rhsmax,
                    rhmin = .data$rhsmin) %>%
      dplyr::select(.data$gcm, .data$date, .data$yr, .data$precip, .data$tmin, .data$tmax, .data$tavg, .data$rhmin, .data$rhmax, .data$units)
    #rename to follow previously written code
  } # close imperial if statement

  if(units == "metric"){
    df <- df1 %>%
      dplyr::mutate(precip = .data$pr,
                    # data source needs to be specified using rlang::.data
                    # error `no visible binding for global variable x` will be thrown
                    tmax = .data$tasmax - 273.1,
                    tmin = .data$tasmin - 273.1,
                    tavg = (.data$tmax + .data$tmin) / 2,
                    gcm = paste(.data$model, .data$rcp, sep = "."),
                    date = as.POSIXlt(.data$date,format="%Y-%m-%d"),
                    yr = lubridate::year(date),
                    units = "metric") %>%
      dplyr::rename(rhmax = .data$rhsmax,
                    rhmin = .data$rhsmin) %>%
      dplyr::select(.data$gcm, .data$date, .data$yr, .data$precip, .data$tmin, .data$tmax, .data$tavg, .data$rhmin, .data$rhmax, .data$units)
    #rename to follow previously written code
  } # close metric if statement

  ifelse(directory == "tempdir()", print("Files have been saved to temporary directory and will be deleted when this R session is closed. To save locally, input where to save them into the `directory` argument."),
         readr::write_csv(df, here::here(SiteID,
                                         paste0(SiteID, ".csv"))))

  # Remove saved climate files
  if(Remove_files == "Y") {
    do.call(file.remove, list(list.files(paste(directory,SiteID,sep="/"), full.names = TRUE)))
    print("Files removed")

    return(df)

  } else {print("Files remain")}
} #close function
